x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/aggregations/pca/PCAStatsResults.java

```diff
@@ -0,0 +1,157 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+package org.elasticsearch.xpack.ml.aggregations.pca;
+
+import org.ejml.data.Complex_F64;
+import org.ejml.data.DMatrixRMaj;
+import org.ejml.dense.row.factory.DecompositionFactory_DDRM;
+import org.ejml.interfaces.decomposition.EigenDecomposition_F64;
+import org.elasticsearch.ElasticsearchException;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.search.aggregations.matrix.stats.MatrixStatsResults;
+import org.elasticsearch.search.aggregations.matrix.stats.RunningStats;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Map;
+
+public class PCAStatsResults extends MatrixStatsResults {
+    /** eigen values from PCA */
+    protected final Map<String, Complex_F64> eigenVals;
```

Comment:
The imaginary part should be identically zero for covariance matrix, it's Hermitian. I'd be tempted to store them as Double in eigenVals.

Reply:
:+1: